tests/components/shelly/test_config_flow.py

```diff
@@ -20,11 +21,16 @@
     "name": "shelly1pm-12345",
     "properties": {"id": "shelly1pm-12345"},
 }
+MOCK_CONFIG = {"domain": "Shelly", "coap_port": 1234}
 
 
 async def test_form(hass):
     """Test we get the form."""
     await setup.async_setup_component(hass, "persistent_notification", {})
+
+    config = MOCK_CONFIG.copy()
+    await async_setup(hass, config)
```

Comment:
When you add a new option and you need to change the set up in the tests, it means that it probably won't work normally?

Because Shelly is not set up when the first config flow is started, there is no COAP port saved. Instead of always pulling it out of `hass.data`, check if `DOMAIN` is in `hass.data`, otherwise just use the default port.

Reply:
I was told that `async_setup` is called by core before starting the flow, and trying on a live HA instance this is true as code worked fine with and without the yaml key.
Only the pytest was failing.

Anyway, fine for me to change the code as suggested.